hbase_storage/src/main/java/com/splicemachine/si/data/hbase/coprocessor/SIObserver.java

```diff
@@ -508,4 +510,22 @@ protected boolean aclCheck(Scan scan) throws IOException, StandardException {
             }
         }
     }
+
+    @Override
+    public void preWALRestore(ObserverContext<? extends RegionCoprocessorEnvironment> ctx, RegionInfo info, WALKey logKey, WALEdit logEdit) throws IOException {
+
```

Comment:
Please add a comment here explaining why this is needed, or at least a pointer to 
DB-9895 and add the full explanation there.

Reply:
I had a hard time reproducing this manually. I can only verify that the token is removed and rows are not purged during WAL replay. Will prepare a parcel to run Stephanie's test on cloud